src/rust/pal/functions.rs

```diff
@@ -3,10 +3,36 @@
 
 use crate::pal::data_structures::SockAddrIn;
 
+#[cfg(all(feature = "catnip-libos", target_os = "windows"))]
+use windows::Win32::Foundation::CHAR;
+
+#[cfg(all(feature = "catnip-libos", target_os = "windows"))]
+use windows::Win32::Networking::WinSock::IN_ADDR;
+
+#[cfg(all(feature = "catnip-libos", target_os = "windows"))]
+use windows::Win32::Networking::WinSock::IN_ADDR_0;
+
+#[cfg(all(feature = "catnip-libos", target_os = "linux"))]
+use libc::in_addr;
+
 //======================================================================================================================
 // Windows functions
 //======================================================================================================================
 
+#[cfg(all(feature = "catnip-libos", target_os = "windows"))]
+pub fn create_sin_addr(octets: &[u8; 4]) -> IN_ADDR {
+    IN_ADDR {
+        S_un: (IN_ADDR_0 {
+            S_addr: u32::from_le_bytes(*octets),
+        }),
+    }
+}
+
+#[cfg(all(feature = "catnip-libos", target_os = "windows"))]
+pub fn create_sin_zero() -> [CHAR; 8] {
+    [CHAR(0); 8]
+}
```

Comment:
This is not specific to Windows, thus it should go in the PAL module, but not inside the "platform-dependent" submodule.

Reply:
Hi @ppenna, this is indeed Windows specific code (notice the CHAR(0)). For Linux, this code is just 0. Or maybe I'm not getting your comment.

Reply:
@anandbonde, I do understand that this code is only used in Windows. But fundamentally it is not specific to this platform, thus you could place it in a PAL layer that is common to both. Rationale:
- In the PAL windows module we find only platform-specific code.
- If Linux starts requiring this to be defined in the future we don't end up with duplicates
- If we introduce support for other LibOS that does require the same definition, we already have it.

Reply:
@anandbonde I got your point. I was reading `CHAR(0)` as u8.